charts/studio/values.yaml

```diff
@@ -1016,3 +1016,340 @@ serviceAccount:
+          name: studio-vector-agent-token
+          key: token
+    - name: KUBERNETES_NODE_IP
+      valueFrom:
+        fieldRef:
+          fieldPath: status.hostIP
+
+  extraObjects:
+    - apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: studio-vector-agent-extended
+        annotations:
+          helm.sh/hook: pre-install
+          helm.sh/hook-delete-policy: hook-failed, before-hook-creation
+
+      rules:
+        - apiGroups:
+            - ""
+          resources:
+            - nodes/metrics
+            - nodes/stats
+          verbs:
+            - get
+
+    - apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRoleBinding
+      metadata:
+        name: studio-vector-agent-extended
+        annotations:
+          helm.sh/hook: pre-install
+          helm.sh/hook-delete-policy: hook-failed, before-hook-creation
+      subjects:
+        - kind: ServiceAccount
+          name: studio-vector-agent
+          namespace: default # FIXME: should be configurable
+      roleRef:
+        kind: ClusterRole
+        name: studio-vector-agent-extended
+
+    - apiVersion: v1
+      kind: Secret
+      metadata:
+        name: studio-vector-agent-token
+        annotations:
+          kubernetes.io/service-account.name: studio-vector-agent
+      type: kubernetes.io/service-account-token
+
+  # defaultVolumes:
+  #   - name: var-log
+  #     hostPath:
+  #       path: "/var/log/"
+
+  # defaultVolumeMounts:
+  #   - name: var-log
+  #     mountPath: "/var/log/"
+  #     readOnly: true
+
+# -- Vector Aggregator configuration for log aggregation and processing
+vector-aggregator:
+  # -- Vector Aggregator enabled
+  enabled: false
+  # -- Vector Aggregator name override
+  fullnameOverride: studio-vector-aggregator
+
+  # -- Deploy as StatefulSet for aggregation and persistence
+  role: Aggregator
+
+  # -- Vector Aggregator replica count
+  replicaCount: 1
+
+  image:
+    # -- The base to use for Vector's image.
+    base: alpine
+
+  # -- Vector Aggregator service account
+  serviceAccount:
+    create: true
+    name: studio-vector-aggregator
+
+  # -- Vector Aggregator resources
+  resources:
+    limits:
+      memory: 512Mi
+    requests:
+      cpu: 200m
+      memory: 256Mi
+
+  # -- Vector Aggregator service configuration
+  service:
+    enabled: true
+    type: ClusterIP
+    ports:
+      - name: logs
+        port: 6000
+        targetPort: 6000
+        protocol: TCP
+      - name: api
+        port: 8686
+        targetPort: 8686
+        protocol: TCP
+
+  # -- Vector Aggregator persistence configuration 
+  persistence:
+    enabled: true
+    storageClass: ""
+    accessModes: ["ReadWriteOnce"]
+    size: 64Gi
+
+  # -- Vector command.
+  command: [/bin/sh, -c]
+
+  # -- Vector arguments.
+  args:
+    - |-
+      while sleep 60; do find /data/vector/logs -type f -mtime +7 -delete; done &
+      exec /usr/local/bin/vector --config-dir /etc/vector/
```

Comment:
Yes, that's the only way I have found to set a retention period of 1 week for logs.